src/tools/rados/rados.cc

```diff
@@ -3284,12 +3268,40 @@ int main(int argc, const char **argv)
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
 
-  global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT, CODE_ENVIRONMENT_UTILITY, 0);
-  common_init_finish(g_ceph_context);
-
   std::map < std::string, std::string > opts;
   std::vector<const char*>::iterator i;
+  std::vector<const char*>::iterator j;
   std::string val;
+
+  /* Necessary to support usage of -f for formatting,
+   * since global_init will remove the -f using ceph
+   * argparse functions. */
+  for (j = args.begin(); j != args.end(); ) {
+    if (strcmp(*j, "--") == 0) {
+      break;
+    } else if ((j+1) == args.end()) {
+      // This can't be a formatting call (no format arg)
+      break; 
+    } else if (strcmp(*j, "-f") == 0) {
+      val = *(j+1);
+
+      Formatter *formatter = NULL;
+      formatter = Formatter::create(val.c_str());
```

Comment:
Formatter *formatter = Formatter::create...

No reason to leave it uninitialized.